### PR TITLE
Add SSO and Authorization.

### PR DESCRIPTION
**Pull request by angelcaamal**
Add SSO and Authorization.

### CODE REVIEW

File: generators/app/templates/app/policies/Authentication.ts [hunk 1]
```diff
@@ -0,0 +1,349 @@
+import { NextFunction, Request, Response } from "express";
```
Comment (rodksquare) [inline generators/app/templates/app/policies/Authentication.ts:1]: @angelcaamal There where important improvements done to this part, including adding custom permission support and a refactor, please check the **develop** branch of https://github.com/rodksquare/ks-cicd-minimal-backend-template/blob/develop/app/policies/Authorization.ts

**Review by rodksquare — CHANGES_REQUESTED** [review]
Please see the changes done on the **develop** branch of https://github.com/rodksquare/ks-cicd-minimal-backend-template and make sure to include them here

File: generators/app/templates/app/policies/Authentication.ts (hunk 1 above)
Comment (angelcaamal) [reply generators/app/templates/app/policies/Authentication.ts:1]: @rodksquare  Thanks, my mistake was taking master branch as example that does not have the latest changes. Working on it.

File: generators/app/templates/package.json.template [hunk 2]
```diff
@@ -49,6 +49,10 @@
     "chai": "4.2.0",
     "del": "3.0.0",
     "eslint": "^6.7.2",
+    "@the-ksquare-group/passport-microsoft": "^0.1.1",
+    "@types/passport-microsoft": "0.0.0",
+    "passport": "^0.4.1",
```
Comment (rodksquare) [inline generators/app/templates/package.json.template]: passport and passport-google-oauth20 should go in dependencies, not dev

Comment (rodksquare) [reply generators/app/templates/package.json.template]: Same for @the-ksquare-group/passport-microsoft

**Review by rodksquare — APPROVED** [review]

**Pull request merged**